components/field/src/Field.js

```diff
@@ -0,0 +1,154 @@
+/** @jsx jsx */
+
+import { jsx, useBrand, overrideReconciler, useInstanceId } from '@westpac/core';
+import { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import { defaultField } from './overrides/field';
+import { defaultLabel } from './overrides/label';
+import { defaultHint } from './overrides/hint';
+import { defaultError } from './overrides/error';
+
+import pkg from '../package.json';
+
+// ==============================
+// Component
+// ==============================
+
+export const Field = ({
+	instanceIdPrefix,
+	label,
+	hideLabel,
+	hint,
+	error,
+	children,
+	overrides: componentOverrides,
+	...rest
+}) => {
+	const {
+		OVERRIDES: { [pkg.name]: tokenOverrides },
+		[pkg.name]: brandOverrides,
+	} = useBrand();
+
+	const defaultOverrides = {
+		Field: defaultField,
+		Label: defaultLabel,
+		Hint: defaultHint,
+		Error: defaultError,
+	};
+
+	const [fieldInstance] = useState(useInstanceId());
+	const [instanceId, setInstanceId] = useState(instanceIdPrefix);
+	const hintId = `gel-field-hint-${fieldInstance};`;
+
+	useEffect(() => {
+		if (!instanceIdPrefix) {
+			setInstanceId(`gel-field-${fieldInstance}`);
+		}
+	}, [instanceIdPrefix]);
+
+	const state = {
+		fieldInstance,
+		instanceId,
+		hintId,
+		label,
+		hideLabel,
+		hint,
+		error,
+		overrides: componentOverrides,
+		...rest,
+	};
+
+	const {
+		Field: { component: Field, styles: fieldStyles, attributes: fieldAttributes },
+		Label: { component: Label, styles: labelStyles, attributes: labelAttributes },
+		Hint: { component: Hint, styles: hintStyles, attributes: hintAttributes },
+		Error: { component: Error, styles: errorStyles, attributes: errorAttributes },
+	} = overrideReconciler(defaultOverrides, tokenOverrides, brandOverrides, componentOverrides);
+
+	const inputProps = {
+		id: instanceId,
+		'aria-describedby': hintId,
+		'aria-invalid': !!error,
+	};
+
+	return (
+		<Field state={state} {...fieldAttributes(state)} css={fieldStyles(state)} {...rest}>
+			<Label
+				htmlFor={instanceId}
+				state={state}
+				{...labelAttributes(state)}
+				css={labelStyles(state)}
+			>
+				{label}
+			</Label>
+			{hint && (
+				<Hint state={state} {...hintAttributes(state)} css={hintStyles(state)}>
+					{hint}
+				</Hint>
+			)}
+			{error && (
+				<Error message={error} state={state} {...errorAttributes(state)} css={errorStyles(state)} />
+			)}
+			{children(inputProps)}
```

Comment:
You can also do this via mapping over the Children so that the composition isn't impacted. We did this in the List component etc. Might be best to stick to the same thing?

Reply:
@dominikwilkowski not sure what you mean here. Im passing the input props to the children so they become available as render props.

Reply:
All you need is making sure the children have the `inputProps` available to them right? That's the same problem in `List` where we need `nested` in each child. The difference to your solution is that children have to be written as functions now (render props) and not composed as normal children. The change is not necessary and might be confusing to our users.